### ISSUE TITLE
No errors ouptutted in the node API when there are type errors, compilation doesn't fail

### ISSUE THREAD

**Issue by marcofugaro** [TypeStrong/fork-ts-checker-webpack-plugin#108]
I'm trying to use this plugin with `thread-loader` as described [in this example](https://github.com/TypeStrong/ts-loader/tree/master/examples/thread-loader), but running it from node instead of the cli, it appears to totally ignore the type checking errors.

Here is how I'm using it, I'm using webpack4:

```js
const compiler = webpack(config)
compiler.hooks.done.tap('done', stats => {
  console.log(stats.hasErrors()) // --> false
})

const devServer = new WebpackDevServer(compiler, devServerConfig)
```

these are the type errors printed by the fork-ts-checker-webpack-plugin plugin:

<img width="1153" alt="screen shot 2018-03-28 at 19 47 22" src="https://user-images.githubusercontent.com/7217420/38046878-0d686452-32c1-11e8-9bc7-1a33e256fb97.png">

There are no errors in the browser and the code is transpiled normally.

**Comment by johnnyreilly** [comment]
Maybe look at this example for reference? https://github.com/TypeStrong/ts-loader/tree/master/examples/thread-loader

**Comment by marcofugaro** [comment]
Yes that's the example I pointed out, forgot to link it sorry 😄

**Comment by johnnyreilly** [comment]
So you're good now?

**Comment by marcofugaro** [comment]
Nope, if running that example using webpack's node api, there are no errors in dev mode

**Comment by johnnyreilly** [comment]
Have you tried using the `async: false` option? The plugin normally bypasses webpack for speed of reporting. This option makes it use webpack

**Comment by marcofugaro** [comment]
Here it is! That's the option I was looking for.

Yes I am using `webpack-dev-server`'s overlay functionality.

From the readme:
> Another scenario where you might want to set this to `false` is if you use the `overlay` functionality of `webpack-dev-server`.

I think it can be written in a better and more explicit way, I read the readme and didn't get that the first time.

You're the man!

**Comment by johnnyreilly** [comment]
Happy to help! If you think you can improve the docs then have a go. Every little helps!
